ballista/rust/core/src/serde/physical_plan/mod.rs

```diff
@@ -799,11 +803,13 @@ impl AsExecutionPlan for PhysicalPlanNode {
         } else if let Some(exec) = plan.downcast_ref::<AggregateExec>() {
             let groups = exec
                 .group_expr()
+                .expr()
                 .iter()
                 .map(|expr| expr.0.to_owned().try_into())
                 .collect::<Result<Vec<_>, BallistaError>>()?;
             let group_names = exec
                 .group_expr()
+                .expr()
                 .iter()
                 .map(|expr| expr.1.to_owned())
                 .collect();
```

Comment:
Here we need to serialize/deserialize the `null_expr` and `groups` as well. This would change the structure of `GROUPING SET` queries. I had to do the same changes on our downstream fork. This seems to work: https://github.com/coralogix/arrow-ballista/commit/68aecb8e7e16ce7346dac3192153e000d3f0dfe4